src/main/java/edu/cornell/kfs/sys/util/CuOjbUtils.java

```diff
@@ -0,0 +1,64 @@
+package edu.cornell.kfs.sys.util;
+
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public final class CuOjbUtils {
+
+    public static Stream<Object[]> buildStreamForReportQueryResults(Iterator<?> reportQueryIterator) {
+        return buildStreamForQueryResults(Object[].class, reportQueryIterator);
+    }
+
+    public static <T> Stream<T> buildStreamForQueryResults(Class<T> queryResultType, Iterator<?> queryIterator) {
+        Objects.requireNonNull(queryResultType, "queryResultType cannot be null");
+        Objects.requireNonNull(queryIterator, "queryIterator cannot be null");
+        return StreamSupport.stream(() -> new QuerySpliterator<>(queryResultType, queryIterator), 0, false);
+    }
+
+    private static class QuerySpliterator<T> implements Spliterator<T> {
```

Comment:
What is the main reason for using a Spliterator here? Are you trying to improve a performance issue?

Reply:
In the DAO class, I was trying to use a Stream to convert the query results into a Map. The results are returned as an Iterator, though, and I haven't found any Java methods to directly convert an Iterator to a Stream. That's why I wrote this Spliterator implementation.

As an alternative, I found some Java examples online that use a lambda expression to create an Iterable object from an Iterator, and that lambda-created Iterable can then be used to create the needed Spliterator for integrating with Streams. I could update this code to switch over to that, so that we don't need to write the Spliterator ourselves.

Reply:
Thank you for clarifying this Chad. It sounds like the new example you have found may simplify the implementation.

Reply:
Ok, I'll rewrite this to use that simplification. Thanks for the input.

Reply:
After further analysis, I noticed that Java has some helper methods for wrapping an Iterator as a Spliterator (which can then be converted into a Stream). I'll update the code to use this strategy.